TCC.Lib/TarCompressCrypt.cs

```diff
@@ -178,15 +178,39 @@ private static string CompressCommand(Block block, TccOption option, ExternalDep
                 case PasswordMode.None:
                     // tar -c C:\SourceFolder | lz4.exe -1 - compressed.tar.lz4
                     cmd = $"{ext.Tar().Escape()} -c {block.Source.Escape()}";
-                    cmd += $" | {ext.Lz4().Escape()} -1 -v - {block.DestinationArchive.Escape()}";
+
+                    switch (option.Algo)
+                    {
+                        case CompressionAlgo.Lz4:
+                            cmd += $" | {ext.Lz4().Escape()} -1 -v - {block.DestinationArchive.Escape()}";
+                            break;
+                        case CompressionAlgo.Brotli:
+                            cmd += $" | {ext.Brotli().Escape()} - -o {block.DestinationArchive.Escape()}";
+                            break;
+                        case CompressionAlgo.Zstd:
+                            break;
+                        default:
+                            throw new ArgumentOutOfRangeException();
+                    }
                     break;
                 case PasswordMode.InlinePassword:
                 case PasswordMode.PasswordFile:
                 case PasswordMode.PublicKey:
                     string passwdCommand = PasswordCommand(option, block);
                     // tar -c C:\SourceFolder | lz4.exe -1 - | openssl aes-256-cbc -k "password" -out crypted.tar.lz4.aes
                     cmd = $"{ext.Tar().Escape()} -c {block.Source.Escape()}";
-                    cmd += $" | {ext.Lz4().Escape()} -1 -v - ";
+                    switch (option.Algo)
+                    {
+                        case CompressionAlgo.Lz4:
+                            cmd += $" | {ext.Lz4().Escape()} -1 -v - ";
+                            break;
+                        case CompressionAlgo.Brotli:
+                            cmd += $" | {ext.Brotli().Escape()} - ";
+                            break;
+                        default:
+                            throw new ArgumentOutOfRangeException();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use a constructor overloads that allows a more meaningful exception message to be provided. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3928)